### PR TITLE
I18n & l10n for Japanese

### PR DESCRIPTION
**Pull request by hanazuki**
Objective: provide the users a way to choose their language preference.

Impelemented features:
- UI internationalization -- Web UI & mails
  - All messages are now translatable.
- Japanese localization
  - Messages are translated into Japanese.
- Locale chooser at the page footer
  - Users can choose their language preference from web UI. 
- Default locale depending on browser locale setting
## ToDo
- [x] I18n for messages, datetimes, etc.
- [x] L10n for Japanese
- [x] Tests??

### CODE REVIEW

**Comment by hanazuki** [comment]
My considerations on testing:
- We need tests for newly introduced features -- user's default locale and the locale chooser.
- Any integration tests should not depend on the current locale, but in what locale shall we run the test suit -- all, any, or default one?
- Mail contents need a test for each locale since they are generated by per-locale views, such as  `password_resets.ja.html.rb`.
- Check for missing translations?

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -2,11 +2,11 @@ class UserMailer < ApplicationMailer
...
   def account_activation(user)
     @user = user
-    mail to: user.email, subject: 'Account activation'
+    mail to: user.email
   end
 
   def password_reset(user)
     @user = user
-    mail to: user.email, subject: 'Password reset'
+    mail to: user.email
```
Comment (alotofwe) [inline app/mailers/user_mailer.rb:10]: subjectを指定しないと翻訳にチェックしに行くの，謎便利機能だけどすごい

**Comment by alotofwe** [comment]
06/30のコミット分まで見たけどよさそうに見える

File: app/mailers/user_mailer.rb (hunk 1 above)
Comment (hanazuki) [inline app/mailers/user_mailer.rb:10]: 便利だけれど，この手の便利機能を多用するとあとあと翻訳メッセージをどこで使っているのかわからなくなったりもしそう．

**Pull request merged**